dtao.c

```diff
@@ -563,7 +760,12 @@ main(int argc, char **argv)
 		BARF("Failed to create display");
 
 	struct wl_registry *registry = wl_display_get_registry(display);
-	wl_registry_add_listener(registry, &registry_listener, NULL);
+
+	struct input_state *istate = malloc(sizeof(struct input_state));
```

Comment:
use calloc() and also `istate` it's never freed.

Reply:
why calloc? is the zero initialization important